src/pycsou/abc/operator.py

```diff
@@ -1867,7 +1909,7 @@ def __array__(self, dtype: typ.Optional[type] = None) -> np.ndarray:
             dtype = pycrt.getPrecision().value
         return self.asarray(xp=np, dtype=dtype)
 
-    def gram(self) -> "LinOp":
+    def gram(self) -> "SquareOp":
```

Comment:
The output should actually be of type ``SelfAdjointOp`` (subclass of ``SquareOp``). Also don't forget to update the output type in the docstring.